pkg/scheduler/core/util.go

```diff
@@ -68,22 +72,46 @@ func calAvailableReplicas(clusters []*clusterv1alpha1.Cluster, spec *workv1alpha
 		return availableTargetClusters
 	}
 
-	// Get the minimum value of MaxAvailableReplicas in terms of all estimators.
+	// get estimators and sort it by priority.
 	estimators := estimatorclient.GetReplicaEstimators()
+	sort.Slice(estimators, func(i, j int) bool {
+		return estimators[i].GetPriority() > estimators[j].GetPriority()
+	})
+
+	priorityOfLastAvailableEstimator := estimatorclient.EstimatorPriority(math.MinInt32)
```

Comment:
I think the solution is a little bit complicated. Here is a simple solution:

1. Group the estimators with different priorities, e.g., 
[{estimators:[es1, es3], priority:9}, {estimators:[es2, es4], priority:8}, ...]
2. Loop these estimators group, do the same previous caculation. if the group is returned, skip the rest ones.

Reply:
Thanks for the suggestion, but I don't think it's getting any better, it leads to one more structure definition and one more layer of loop iteration.

Reply:
Complex structures or more loops can also complicate the problem.

Reply:
> Group the estimators with different priorities, e.g., [{estimators:[es1, es3], priority:9}, {estimators:[es2, es4], priority:8}, ...]. Loop these estimators group, do the same previous caculation. if the group is returned, skip the rest ones.

done, I have applied this strategy.

the nits is there are triple nested for loop now.